funnel/views/mixins.py

```diff
@@ -130,3 +133,75 @@ def loader(self, profile, project, section):
             ).first_or_404()
         g.profile = section.project.profile
         return section
+
+
+class DraftModelViewMixin(object):
+    def get_draft(self, current_obj=None):
+        """
+        Returns the draft object for `current_obj`. Defaults to `self.obj`.
+        `current_obj` is needed in case of multi-model views.
+        """
+        obj = current_obj or self.obj
+        return Draft.query.get((self.model.__tablename__, obj.uuid))
+
+    def delete_draft(self, current_obj=None):
+        """
+        Deletes draft for `current_obj`, or `self.obj` if `current_obj` is `None`.
+        """
+        draft = self.get_draft(current_obj)
+        if draft is not None:
+            db.session.delete(draft)
+            db.session.commit()
+
+    def autosave_get(self, current_obj=None):
+        """
+        Returns a tuple of the current draft revision and the formdata needed to initialize forms
+        """
+        draft = self.get_draft(current_obj)
+        initial_formdata = MultiDict(draft.body['form']) if draft is not None else None
+        return draft.revision, initial_formdata
```

Comment:
If this method is not handling a GET request (as the `_get` suffix suggests), it should not be called `autosave_get`. This should be `get_draft_data`.